### PR TITLE
examples added

### PR DESCRIPTION
**Pull request by bvbasavaraju**
While going though the library, I wanted to practice with the trial and error. So, I used the same examples what was mentioned in read me file but as a build-able project with some small tweaks for my better understanding.

I followed the guidelines as suggested, please let me know if I have missed anything! 

I was not sure if the branch should be feature or bugfix. for sure it was not a bugfix. So, i selected feature even though it is just an example. I hope that is fine.

### CODE REVIEW

**Comment by lukevalenty** [comment]
> @lukevalenty : one question related to code style!
> 
> 
> 
> I used clang-format tool to format the code. is there any recommendation!? 
> 
> 
> 
> Also, struct name has to be snake_case only!? I saw MainLoop component in CamelCase! Could you please help me understand?

Hi @bvbasavaraju! Thanks for the pull request. :) 

There are some differences in how clang-format works from one version to the next. Right now we are stuck on version 11. You'll need to make sure to run clang-format with that specific version.

We previously used CamelCase for our type names and are now using snake_case only. Their might be some outliers in documentation or comments that haven't been updated yet. 😬

**Comment by bvbasavaraju** [comment]
> 

Hi @lukevalenty ,

thank you for the update. I used the clang-format version "14.0.6" and I can see that "[quality_checks_pass](https://github.com/intel/compile-time-init-build/actions/runs/3574425708/jobs/6009675030)" has been passed for the style.

Could you please let me know if I still need to rerun with clang-format version 11 and update my PR if it is required!?

Thank you.

**Comment by elbeno** [comment]
@bvbasavaraju If the quality checks passed that is all good, no need to re-run with clang-format-11.  Minor changes are possible between clang-format versions, often on newer C++ features as clang-format learns to format them better. But in this case it seems 11 and 14 agree. Thanks for the contribution!

File: examples/cib_serial_port/main.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#include <cib/cib.hpp>
+
+#include <iostream>
+
+// Services - basically like a function prototypes - callback function
+// declaration!!
```
Comment (lukevalenty) [inline examples/cib_serial_port/main.cpp:6]: suggestion: reserve exclamation marks `!` for critically important things so they stand out. nearly every sentence should end in a period `.`

File: examples/cib_serial_port/main.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+struct run_t : public cib::callback_meta<> {};
+
+// components defintion!!
+struct core_component_t {
+    constexpr static auto config = cib::config(
+
+        cib::exports<send_byte_t>,
+
+        cib::extend<send_byte_t>([](uint8_t byte) {
+            std::cout << "From CoreComponent: " << byte << std::endl;
```
Comment (lukevalenty) [inline examples/cib_serial_port/main.cpp:22]: suggest: update the names of compoments in strings to match their snake_case name

**Review by lukevalenty — CHANGES_REQUESTED** [review]
thanks for your pull request, these are some helpful examples for people looking to see how to use the framework. :)

i'm requesting a minor change to the code comment grammar, just replace `!!` with `.` and we should be good to merge this :)

File: examples/cib_serial_port/main.cpp (hunk 1 above)
Comment (bvbasavaraju) [reply examples/cib_serial_port/main.cpp]: Thank you for the input. I have updated the same.

File: examples/cib_serial_port/main.cpp (hunk 2 above)
Comment (bvbasavaraju) [reply examples/cib_serial_port/main.cpp]: Thank you for the suggestion. I have replaced with the snake_case name.

**Comment by bvbasavaraju** [comment]
> @bvbasavaraju If the quality checks passed that is all good, no need to re-run with clang-format-11. Minor changes are possible between clang-format versions, often on newer C++ features as clang-format learns to format them better. But in this case it seems 11 and 14 agree. Thanks for the contribution!

@elbeno thank you for the guidance.

**Comment by bvbasavaraju** [comment]
> thanks for your pull request, these are some helpful examples for people looking to see how to use the framework. :)
> 
> i'm requesting a minor change to the code comment grammar, just replace `!!` with `.` and we should be good to merge this :)

@lukevalenty I have worked on review comment and suggestion. Please let me know if additional changes are needed

**Review by lukevalenty — APPROVED** [review]
looks good :)

**Comment by bvbasavaraju** [comment]
> looks good :)

Sure thank you.

**Pull request merged**